src/main/java/seedu/eduke8/parser/MenuParser.java

```diff
@@ -108,74 +106,40 @@ public Command parseCommand(DisplayableList topicList, String userInput) {
-                        topicName = commandArr[2].substring(
-                                commandArr[2].indexOf(TOPIC_INDICATOR) + LENGTH_OF_TOPIC_INDICATOR);
-                        userTimer = Integer.parseInt(commandArr[1].substring(
-                                commandArr[1].indexOf(TIMER_INDICATOR) + LENGTH_OF_TIMER_INDICATOR));
-                    }
-
-                } else if (commandArr[1].contains(NUMBER_OF_QUESTIONS_INDICATOR)) {
-                    if (commandArr[2].contains(TIMER_INDICATOR)) {
-                        numOfQuestions = Integer.parseInt(commandArr[1].substring(
-                                commandArr[1].indexOf(NUMBER_OF_QUESTIONS_INDICATOR) + LENGTH_OF_QUESTIONS_INDICATOR));
-                        topicName = commandArr[3].substring(
-                                commandArr[3].indexOf(TOPIC_INDICATOR) + LENGTH_OF_TOPIC_INDICATOR);
-                        userTimer = Integer.parseInt(commandArr[2].substring(
-                                commandArr[2].indexOf(TIMER_INDICATOR) + LENGTH_OF_TIMER_INDICATOR));
-
-                    } else if (commandArr[2].contains(TOPIC_INDICATOR)) {
-                        numOfQuestions = Integer.parseInt(commandArr[1].substring(
-                                commandArr[1].indexOf(NUMBER_OF_QUESTIONS_INDICATOR) + LENGTH_OF_QUESTIONS_INDICATOR));
-                        topicName = commandArr[2].substring(
-                                commandArr[2].indexOf(TOPIC_INDICATOR) + LENGTH_OF_TOPIC_INDICATOR);
-                        userTimer = Integer.parseInt(commandArr[3].substring(
-                                commandArr[3].indexOf(TIMER_INDICATOR) + LENGTH_OF_TIMER_INDICATOR));
+                boolean isUserTimerSet = false;
+                boolean isTopicNameSet = false;
+                boolean isNumOfQuestionsSet = false;
+
+                for (int i = 1; i < 4; i++) {
+                    if (commandArr[i].contains(TIMER_INDICATOR) && !isUserTimerSet) {
+                        userTimer = Integer.parseInt(commandArr[i].substring(
+                                commandArr[i].indexOf(TIMER_INDICATOR) + LENGTH_OF_TIMER_INDICATOR));
+                        isUserTimerSet = true;
+                    } else if (commandArr[i].contains(TOPIC_INDICATOR) && !isTopicNameSet) {
+                        topicName = commandArr[i].substring(
+                                commandArr[i].indexOf(TOPIC_INDICATOR) + LENGTH_OF_TOPIC_INDICATOR);
+                        isTopicNameSet = true;
+                    } else if (commandArr[i].contains(NUMBER_OF_QUESTIONS_INDICATOR) && !isNumOfQuestionsSet) {
+                        numOfQuestions = Integer.parseInt(commandArr[i].substring(
+                                commandArr[i].indexOf(NUMBER_OF_QUESTIONS_INDICATOR) + LENGTH_OF_QUESTIONS_INDICATOR));
+                        isNumOfQuestionsSet = true;
                     }
+                }
```

Comment:
Well thought out! Nice implementation which integrates the conditionals from before.